wetlab/views.py

```diff
@@ -2441,230 +2410,119 @@ def record_samples(request):
                 return render(
                     request,
                     "wetlab/record_sample.html",
                     {
                         "fields_info": fields_info,
-                        "error_message": core.utils.load_batch.validate_header(
-                            sample_batch_df
-                        ),
+                        "error_message": validation_result["error_message"],
                     },
                 )
-
-            # Test if date columns have date format
-            if core.utils.load_batch.check_format_date(sample_batch_df):
+            # format date columns
+            sample_batch_df = core.utils.load_batch.check_and_format_date(
+                sample_batch_df
+            )
+            if not isinstance(sample_batch_df, pd.DataFrame):
                 return render(
                     request,
                     "wetlab/record_sample.html",
                     {
                         "fields_info": fields_info,
-                        "error_message": core.utils.load_batch.check_format_date(
-                            sample_batch_df
-                        ),
+                        "error_message": sample_batch_df,
                     },
                 )
-
-            # Reformat date columns to correct format (mandatory before converting to json)
-            sample_batch_df = core.utils.load_batch.format_date(sample_batch_df)
-
+            sample_batch_df = core.utils.load_batch.heading_refactor(sample_batch_df)
             # Convert pandas dataframe to json list of dictionaries
             batch_json_data = core.utils.load_batch.read_batch_sample_file(
                 sample_batch_df
             )
-
             # validate mandatory and redundant samples
-            validation = core.utils.samples.validate_sample_data(
+            (
+                sample_validation,
+                validation_result,
+            ) = core.utils.samples.validate_sample_data(
                 batch_json_data, req_user, __package__
             )
-
-            # If all samples are validated check if we need to add project data
-            project = []
-            for sample in batch_json_data:
-                if (
-                    sample["sample_project"] not in project
-                    and sample["sample_project"] != "None"
-                ):
-                    project.append(sample["sample_project"])
-            # If no project just add samples
-            if not project:
-                for val in validation:
-                    if not val["Validate"]:
-                        return render(
-                            request,
-                            "wetlab/record_sample.html",
-                            {
-                                "fields_info": fields_info,
-                                "validation": validation,
-                            },
-                        )
-                try:
```

Comment:
I would not remove this as safety measure, as before

Reply:
yes, my mistake, sorry